### PR TITLE
Remove mock test fallback

### PR DESCRIPTION
**Pull request by engelmi**
## Please describe the change you are making

This PR solves https://github.com/oVirt/go-ovirt-client/issues/191. 

It removes the fallback to the mock client when creating the live client creation fails (e.g. missing env variables or typo in env variable name) during tests. Also, the function to create a test helper with a mock client is exported and the tests were rewritten so that each test is executed with a mock and live client in sub tests. This also enables to explicitly run the tests either with the mocked backend or a live connection. 

## Are you the owner of the code you are sending in, or do you have permission of the owner?

yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

yes

### CODE REVIEW

**Review by eslutsky — COMMENTED** [review]
this is changes the structure of the tests, to run a single test:
the whole path needs to be passed:
for mock:
`-run .*TestImageUpload.*/with_Mock/disk already exists
`
both mock and live:
`-run .*TestImageUpload.*/.*/disk already exists
`
i'm fine with this , but we should add something to the doc about this.

File: README.md [hunk 1]
```diff
@@ -92,11 +92,17 @@ The test helper can work in two ways:
...
-The easiest way to set up the test helper is using environment variables. To do that, you can use the
-`ovirtclient.NewTestHelperFromEnv()` function:
+The ovirtclient.NewMockTestHelper() function can be used to create a test helper with a mock client in the backend:
 
 ```go
-helper := ovirtclient.NewTestHelperFromEnv(ovirtclientlog.NewNOOPLogger())
+helper := ovirtclient.NewMockTestHelper(ovirtclientlog.NewNOOPLogger())
+```
+
+The easiest way to set up tthe test helper for a live connection is by using environment variables. To do that, you
```
[the remark this answers is not in the document]
Comment (eslutsky) [reply README.md:101]: /ost

**Comment by janosdebugs** [comment]
I wonder if the same could be accomplished with the `TestMain` function. It's a bit of magic, but it would make reading the tests a whole lot easier.

**Comment by engelmi** [comment]
> I wonder if the same could be accomplished with the `TestMain` function. It's a bit of magic, but it would make reading the tests a whole lot easier.

Good point. I haven't used `TestMain` yet, though. Will have a look into it.

**Comment by janosdebugs** [comment]
Looks good, let's see what OST says.

**Review by eslutsky — APPROVED** [review]
looks good, but you have few `undo` commits there :) please squash the commits before merging

**Pull request merged**